system/jlib/jptree.cpp

```diff
@@ -3031,19 +3151,24 @@ void CAtomPTree::setAttribute(const char *key, const char *val)
     AttrValue *v = findAttribute(key);
     if (v)
     {
-        if (streq(v->value->get(), val))
+        if (streq(v->value.get(), val))
             return;
         CriticalBlock block(hashcrit);
```

Comment:
efficiency: You could avoid the cs if (!getPtr()) and value.set() succeeds.  Should be relatively easy to cleanly rearrange.